Http1ObjectEncoder.combine may fail if both futures fail for the same reason

With Armeria 1.22.1, on an unstable connection with many timeouts, I am observing the following issue once in a while

```
java.lang.IllegalArgumentException: Self-suppression not permitted
	at java.base/java.lang.Throwable.addSuppressed(Throwable.java:1064)
	at com.linecorp.armeria.internal.common.Http1ObjectEncoder$1.operationComplete(Http1ObjectEncoder.java:156)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
```

Root cause, which is likely not relevant to this issue, is :
```
Caused by: io.netty.handler.ssl.SslClosedEngineException: SSLEngine closed already
	at io.netty.handler.ssl.SslHandler.wrap(SslHandler.java:861)
	... 53 common frames omitted
```

I suppose that `addSuppressed` should be skipped if `firstCause == secondCause`

> Caused by: io.netty.handler.ssl.SslClosedEngineException: SSLEngine closed already
	at io.netty.handler.ssl.SslHandler.wrap(SslHandler.java:861)
	... 53 common frames omitted

I'm curious how often this occurs and which version this warning log started to appear 🤔 
May I ask what the previous version was before upgrading?

--- a/core/src/main/java/com/linecorp/armeria/internal/common/Http1ObjectEncoder.java
+++ b/core/src/main/java/com/linecorp/armeria/internal/common/Http1ObjectEncoder.java
@@ -145,18 +145,15 @@ public abstract class Http1ObjectEncoder implements HttpObjectEncoder {
                     final Throwable firstCause = first.cause();
                     final Throwable secondCause = second.cause();
 
-                    Throwable combinedCause = null;
-                    if (firstCause != null) {
+                    final Throwable combinedCause;
+                    if (firstCause == null) {
+                        combinedCause = secondCause;
+                    } else {
+                        if (secondCause != null && secondCause != firstCause) {
+                            firstCause.addSuppressed(secondCause);
+                        }
                         combinedCause = firstCause;
                     }
-                    if (secondCause != null) {
-                        if (combinedCause == null) {
-                            combinedCause = secondCause;
-                        } else {
-                            combinedCause.addSuppressed(secondCause);
-                        }
-                    }
-
                     if (combinedCause != null) {
                         promise.setFailure(combinedCause);
                     } else {

